documentation/1.2/tour/interop.md

```diff
@@ -85,7 +85,107 @@ Aether. You can find more information [here](../../reference/repository/maven).
+ sibility:=reexport
+Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version>=1.7))"
+```
+The Ceylon module imports are translated into a `Require-Bundle` header, and `shared` imports are translated into a `reexport` directive.The `java.base "7"` import is translated into a `Require-Capability` header that requires the same Java version.
+
+Note that the `com.redhat.ceylon.dist` bundle is systematically and implicitely added as a dependency to all generated Ceylon archives.
+
+However it is also possible to customize the OSGI metadata of the generated archive in 2 ways:
+
+1. Additional manifest entries or resources (such as a desclarative service descriptor) can be added by simply adding resources as explained [below](./interop.md#meta_inf_and_web_inf).
+
+2. Module imports that correspond to __dependencies provided by the OSGI container__ (such as the OSGI framework bundle or any bundle assumed to be available in the OSGI container) should be omitted from the generated `Require-Bundle` header. The `--osgi-provided-bundles=<modules>` option of the `ceylon compile` command allows specifying those modules. The same list can also be specified in the [compiler section](http://www.ceylon-lang.org/documentation/1.2/reference/tool/config#_compiler_section) of the configuration file.
+  
+  Generally, once omitted from the `Require-Bundle` header, the dependencies to provided bundles should be declared in an `Import-Package` header added in the [`MANIFEST.MF` resource](./interop.md#meta_inf_and_web_inf), as mentioned in point 1.
+ 
+#### Ceylon Metamodel registering
+
+In order to be able to fully leverage the power of the Ceylon language, the metamodel should be initialized for each module used by a Ceylon application. It is automatically performed when running from the command line through the `ceylon run` command, but in an OSGi container it should be done when starting the Ceylon module. The Ceylon OSGI distribution provides an easy way to register the metamodel for a module, as well as for all the modules transitively imported.
+
+This can be done by adding an OSGI activator class to the Ceylon module, that performs this metamodel registration. The `com.redhat.ceylon.dist` OSGI bundle, implicitely required by all Ceylon modules, already provides such an activator class: `com.redhat.ceylon.dist.osgi.Activator`
+
+So if you don't need to do anything else in your OSGI Ceylon bundle at startup, you can simply add the following line to your [MANIFEST.MF resource](./interop.md#meta_inf_and_web_inf):
+```
+Bundle-Activator: com.redhat.ceylon.dist.osgi.Activator
+```
+
+
+Alternatively, if you need to perform some sort initialization in your Ceylon module bundle startup, you can also define an `Activator` class that will delegate to the default `com.redhat.ceylon.dist.osgi.Activator`. In this case:
+
+1. Explicitly import the `com.redhat.ceylon.dist` module in your module descriptor:
+```Ceylon
```

Comment:
another broken code block (and once it works, it probably needs to be indented so it doesn’t break the list)